ubuntu/recovery/ubports-mounter.sh

```diff
@@ -2,19 +2,28 @@
 
 DATA_MOUNT_CODE=1
 
+RETRY_COUNTER=0
 while [ "$DATA_MOUNT_CODE" != "0" ]; do
-    mount -t ext4 /dev/block/by-name/userdata /data > /dev/kmsg
+    if [ "$RETRY_COUNTER" -gt "3" ]; then
+        echo "setup_fake_cache.sh - reached maximum number of retries" > /dev/kmsg
+
+        setprop halium.datamount.done 1
+        exit 1
+    fi
+    RETRY_COUNTER=$((RETRY_COUNTER+1))
+
+    mount /dev/block/by-name/userdata /data > /dev/kmsg
     DATA_MOUNT_CODE=$?
-    sleep 1
 done
 
-mkdir /cache > /dev/kmsg
-mkdir /data/android-data > /dev/kmsg
-mkdir /data/android-data/cache > /dev/kmsg
-mkdir /data/android-data/cache/recovery > /dev/kmsg
-mount -o bind /data/android-data/cache /cache > /dev/kmsgs
+mkdir /data/cache/ > /dev/kmsg
+mkdir /data/cache/recovery/ > /dev/kmsg
 
-setprop halium.datamount.done 1
+while [ "$DATA_MOUNT_CODE" == "0" ]; do
```

Comment:
This looks like endless loop to me, as you never set DATA_MOUNT_CODE inside. But why is it even here?

Reply:
Fixed

Reply:
The service will automatically exit after 20 or so seconds, which isn't much of an issue IMO.

Reply:
I don't know why it would exit. Code wise it looks like it will be stuck in while loop forever, so better if you figure out the issue with fake cache mount.

Reply:
Fixed